hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFilePrettyPrinter.java

```diff
@@ -760,6 +830,37 @@ private void printHistogram(Histogram histogram) {
       output.printf(locale, "              99%% <= %2.2f%n", snapshot.get99thPercentile());
       output.printf(locale, "            99.9%% <= %2.2f%n", snapshot.get999thPercentile());
       output.printf(locale, "             count = %d%n", histogram.getCount());
+
```

Comment:
The below block is really unobvious to me how it is (presumably) printing mins and maxes but it may just be late for me. I'll take another look tomorrow. Perhaps a comment of what the process being implemented is could be of assistance?

Reply:
The min/max change is handled above on lines 822/823. The below logic prints the counts-by-range, i.e. the last 4 lines here:

```
   Key length:
               min = 43
               max = 137
              mean = 104.97
            stddev = 25.93
            median = 105.00
              75% <= 137.00
              95% <= 137.00
              98% <= 137.00
              99% <= 137.00
            99.9% <= 137.00
             count = 852277
                0 <= 10
              785 <= 50
           404643 <= 100
           446849 <= 500
```

I could add a comment here just to describe what's happening. It's similar to how we handle range counts in JMX

Reply:
@cbaenziger just pushed a comment, let me know if it clears anything up

Reply:
Thanks! The code and review comments makes it more clear what is intended. I think the output will need some explanation of what is being displayed after the count.

Right now, it looks like it's on lines [864-866](https://github.com/apache/hbase/pull/4638/commits/820104f375217e2cad2cfa0d3f3c4631a6ec6599#diff-35423d30ce8e7b499844db2cd71563dea12517322eb476239bc7867ccd96ac6dL860-L862) printing:
`<count> <= <range>`

While all the other lines print:
`<descriptor> <= <value from data>`

Could the range count be reversed to match the layout of the other lines?
`<range> <= <count>`

Lastly, this may be a question/solution chasing a problem, but will this show a bi-modal set of ranges clearly (e.g. if I have keys of 50 bytes and keys of 5,000 bytes only) or will the elided bounding ranges be needed to point that out? Or is that what [line 845](https://github.com/apache/hbase/pull/4638/commits/820104f375217e2cad2cfa0d3f3c4631a6ec6599#diff-35423d30ce8e7b499844db2cd71563dea12517322eb476239bc7867ccd96ac6dL840-L841) is doing already?